packages/next/src/client/components/navigation.ts

```diff
@@ -182,7 +182,9 @@ function getSelectedLayoutSegmentPath(
     return segmentPath
   }
 
-  segmentPath.push(segmentValue)
+  if (segmentValue !== 'children' && !segmentValue.startsWith('@')) {
```

Comment:
seems like this is a bit too naive -- the segment could be called `children` and now we've broken that case